tf2_ros/test/time_reset_test.cpp

```diff
@@ -27,79 +27,85 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include <chrono>
 #include <gtest/gtest.h>
 #include <tf2_ros/transform_broadcaster.h>
 #include <tf2_ros/transform_listener.h>
-#include <rosgraph_msgs/Clock.h>
+#include <rosgraph_msgs/msg/clock.hpp>
+#include <builtin_interfaces/msg/time.hpp>
+#include <rclcpp/rclcpp.hpp>
 
-using namespace tf2;
+void spin_for_a_second(std::shared_ptr<rclcpp::Node>& node)
+{
+  rclcpp::Rate r(10);
+  rclcpp::spin_some(node);
+  for (uint8_t i = 0; i < 10; ++i)
+  {
+    r.sleep();
+    rclcpp::spin_some(node);
+  }
+}
 
-TEST(tf2_ros_transform_listener, time_backwards)
+TEST(tf2_ros_time_reset_test, time_backwards)
 {
+  std::shared_ptr<rclcpp::Node> node_ = std::make_shared<rclcpp::Node>("transform_listener_backwards_reset");
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
 
-  tf2_ros::Buffer buffer;
+  tf2_ros::Buffer buffer(clock);
   tf2_ros::TransformListener tfl(buffer);
-  tf2_ros::TransformBroadcaster tfb;
+  tf2_ros::TransformBroadcaster tfb(node_);
 
-  ros::NodeHandle nh = ros::NodeHandle();
+  auto clock_pub = node_->create_publisher<rosgraph_msgs::msg::Clock>("/clock", 1);
 
-  ros::Publisher clock = nh.advertise<rosgraph_msgs::Clock>("/clock", 5);
 
-  rosgraph_msgs::Clock c;
-  c.clock = builtin_interfaces::msg::Time(100);
-  clock.publish(c);
+  rosgraph_msgs::msg::Clock c;
+  c.clock = rclcpp::Time(100, 0);
```

Comment:
It's fine to me to switch from `builtin_interfaces::msg::Time` -> `rclcpp::Time`, but I'm just wondering why?

Reply:
there are no constructor for builtin_interfaces::msg::Time(100); or builtin_interfaces::msg::Time(100, 0);